src/Blocks/Callout.php

```diff
@@ -0,0 +1,184 @@
+ * @psalm-import-type FileJson from \Notion\Common\File
+ *
+ * @psalm-type CalloutJson = array{
+ *      callout: array{
+ *          text: RichTextJson[],
+ *          children: array{ type: BlockJson },
+ *          icon: EmojiJson|FileJson,
+ *      },
+ * }
+ */
+class Callout implements BlockInterface
+{
+    private const TYPE = Block::TYPE_CALLOUT;
+
+    private Block $block;
+
+    /** @var \Notion\Common\RichText[] */
+    private array $text;
+
+    private Emoji|File $icon;
+
+    /** @var \Notion\Blocks\BlockInterface[] */
+    private array $children;
+
+    /**
+     * @param \Notion\Common\RichText[] $text
+     * @param \Notion\Blocks\BlockInterface[] $children
+     */
+    private function __construct(
+        Block $block,
+        array $text,
+        Emoji|File $icon,
+        array $children,
+    ) {
+        if (!$block->isCallout()) {
+            throw new \Exception("Block must be of type " . self::TYPE);
+        }
+
+        $this->block = $block;
+        $this->text = $text;
+        $this->icon = $icon;
+        $this->children = $children;
+    }
+
+    public static function create(): self
+    {
+        $block = Block::create(self::TYPE);
+        $icon = Emoji::create("⭐");
+
+        return new self($block, [], $icon, []);
+    }
+
+    public static function fromString(string $emoji, string $content): self
+    {
+        $block = Block::create(self::TYPE);
+        $text = [ RichText::createText($content) ];
+        $icon = Emoji::create($emoji);
+
+        return new self($block, $text, $icon, []);
+    }
+
+    public static function fromArray(array $array): self
+    {
+        /** @psalm-var BlockJson $array */
+        $block = Block::fromArray($array);
+
+        /** @psalm-var CalloutJson $array */
+        $callout = $array[self::TYPE];
+
+        $text = array_map(fn($t) => RichText::fromArray($t), $callout["text"]);
+
+        $iconArray = $callout["icon"];
+        if ($iconArray["type"] === "emoji") {
+            /** @psalm-var EmojiJson $iconArray */
+            $icon = Emoji::fromArray($iconArray);
+        } else {
+            /** @psalm-var FileJson $iconArray */
+            $icon = File::fromArray($iconArray);
+        }
+
+        $children = array_map(fn($b) => BlockFactory::fromArray($b), $callout["children"]);
+
+        return new self($block, $text, $icon, $children);
+    }
+
+    public function toArray(): array
+    {
+        $array = $this->block->toArray();
+
+        $array[self::TYPE] = [
+            "text"     => array_map(fn(RichText $t) => $t->toArray(), $this->text),
+            "icon"     => $this->icon->toArray(),
+            "children" => array_map(fn(BlockInterface $b) => $b->toArray(), $this->children),
+        ];
+
+        return $array;
+    }
+
+    public function toString(): string
+    {
+        $string = "";
+        foreach ($this->text as $richText) {
+            $string = $string . $richText->plainText();
```

Comment:
`return implode('', array_map(static fn (RichText $t): string => $t->plainText(), $this->text));` or such?

Reply:
Yes, same thing. It is only syntax preference or this comes with another benefit?